src/py/_pyodide/_core_docs.py

```diff
@@ -124,6 +124,36 @@ def to_bytes(self) -> bytes:
         an ArrayBuffer view.
         """
 
+    def to_file(self, fd: int):
+        """Writes the entire buffer to a file descriptor.
+
+        Will write the entire contents of the buffer to the current position of
+        the file descriptor.
+
+        Present only if the wrapped Javascript object is an ArrayBuffer or an
+        ArrayBuffer view.
+        """
+
+    def from_file(self, fd: int):
+        """Reads from a file descriptor into the buffer.
+
+        Will try to read a chunk of data of size the length of the buffer from
+        the current position of the file descriptor.
+
+        Present only if the wrapped Javascript object is an ArrayBuffer or an
+        ArrayBuffer view.
+        """
+
+    def _into_file(self, fd: int):
+        """Will write the entire contents of the buffer to the current position
+        of the file descriptor using ``canOwn : true``. If the file is in the
```

Comment:
I had to search emscripten source code to understand what this does. More explanations of what it means won't hurt.

Also couldn't this be a `tofile(fh, can_own=True)`?

Reply:
> couldn't this be a `tofile(fh, can_own=True)`?

It could be, but I would prefer to keep this one as a private API for now because it's a bit "unsafe". In particular, after mounting this `ArrayBuffer` into the file system, the file system "owns" the buffer so file operations on the file could be cached in the wasm memory. Changing the buffer directly can cause data corruption.

Reply:
This `_into_file` command is intended primarily as an optimization for `unpack_archive`. It eliminates another copy compared to `to_file`

#### original code (3 copies)
```py
resp = await pyfetch("archive.zip")
bytes = await resp.bytes() # copy data into wasm heap
f.write(bytes) # copy data into memfs
shutil.unpack_archive(f.name) # copy data from memfs into wasm heap
```

#### to_file (2 copy)
```py
resp = await pyfetch("archive.zip")
buf = await resp.buffer() # no copy
buf.to_file(f) # copy data into memfs
shutil.unpack_archive(f.name) # copy data from memfs into wasm heap
```
vs
#### into_file (1 copy)
```py
resp = await pyfetch("archive.zip")
buf = await resp.buffer() # no copy
buf._into_file(f) # no copy, give memfs ownership of buf
shutil.unpack_archive(f.name) # copy data from memfs into wasm heap
```

Reply:
How's it look now?